src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/EndOfServiceReportController.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller
+
+import org.springframework.http.ResponseEntity
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.LocationMapper
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller.mappers.JwtAuthUserMapper
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.EndOfServiceReportDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.EndOfServiceReportService
+import java.util.UUID
+
+@RestController
+class EndOfServiceReportController(
+  val jwtAuthUserMapper: JwtAuthUserMapper,
+  val locationMapper: LocationMapper,
+  val endOfServiceReportService: EndOfServiceReportService,
+) {
+
+  @PostMapping("/referral/{id}/end-of-service-report")
```

Comment:
Given our intent here is to "start an end of service report for a referral", that relate to similar intents like "start an action plan for a referral", which we implemented as `POST /draft-action-plan`

Should we use that pattern here, too, to be consistent?